### PR TITLE
Refs #3809 - Remove rubocop TODOs

### PR DESCRIPTION
**Pull request by dLobatog**
Removed the following TODOs so that cops for these will run from now on:

Lint/AmbiguousOperator, DefEndAlignment, DeprecatedClassMethods
EnsureReturn, RequireParentheses, Void, BlockAlignment, EndAlignment,
UselessAccessModifier,

Style/IndentArray, IndentHash, Tab, UnlessElse, UnneededPercentQ, UnneededPercentX, RedundantReturn

Rails/Validation

### CODE REVIEW

**Comment by dLobatog** [comment]
[test] It seems like Jenkins picked up [the wrong PR](http://ci.theforeman.org/job/test_develop_pull_request/8637/)

**Comment by domcleal** [comment]
That would be very worrying, looks entirely correct to me though... branch 3809 on your remote.

**Comment by dLobatog** [comment]
[test], only sqlite in 1.9.3 failed...

**Comment by dLobatog** [comment]
Latest update cleans up some new commits that failed to pass the lint and removes a few more. 
It would be great if this could be merged soon, especially because until this gets merged, new commits might Rubocop errors fixed here.

**Comment by dLobatog** [comment]
Rebased against develop again

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -22,16 +22,17 @@ def status
...
-    result[:status] = 200
-    result[:version] = SETTINGS[:version].full
-    result[:db_duration_ms] = ((Time.now - start) * 1000).round.to_s
-  rescue Exception => e
-    result[:result] = 'fail'
-    result[:status] = 500
-    result[:message] = e.message
-  ensure
-    return result
+    begin
```
Comment (ohadlevy) [inline app/controllers/home_controller.rb:25]: why the extra begin? is it to avoid the ensure ?

File: app/controllers/hosts_controller.rb [hunk 2]
```diff
@@ -499,16 +499,15 @@ def process_hostgroup
...
-      host = Host::Base.authorized(:view_hosts, Host).find(params[:host][:id])
-      host = host.becomes Host::Managed
-      host.attributes = params[:host]
-      host
-    else
-      Host.new(params[:host])
-    end
+              host = Host::Base.authorized(:view_hosts, Host).find(params[:host][:id])
+              host = host.becomes Host::Managed
+              host.attributes = params[:host]
```
Comment (ohadlevy) [inline app/controllers/hosts_controller.rb:504]: are we also aligning on = ?

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -210,7 +210,7 @@ def auto_complete_search(name, val, options = {})
   end
 
   def help_path
-    link_to _("Help"), :action => "welcome" if File.exists?("#{Rails.root}/app/views/#{controller_name}/welcome.html.erb")
+    link_to _("Help"), :action => "welcome" if File.exist?("#{Rails.root}/app/views/#{controller_name}/welcome.html.erb")
```
Comment (ohadlevy) [inline app/helpers/application_helper.rb:213]: why change? they are aliased one to another?

File: app/models/compute_resources/foreman/model/rackspace.rb [hunk 4]
```diff
@@ -105,7 +105,6 @@ def client
         :rackspace_auth_url => url,
         :rackspace_region => region.downcase.to_sym
       )
-      return @client
```
Comment (ohadlevy) [inline app/models/compute_resources/foreman/model/rackspace.rb:108]: wonder if this shouldnt be 

``` ruby
@client ||= ....
```

File: app/models/taxonomies/location.rb [hunk 5]
```diff
@@ -14,14 +14,14 @@ class Location < Taxonomy
...
-        user = User.current
-        if user.admin?
-          conditions = { }
-        else
-          conditions = sanitize_sql_for_conditions([" (taxonomies.id in (?))", user.location_and_child_ids])
-        end
-        where(conditions)
-      }
+    user = User.current
+    if user.admin?
```
Comment (ohadlevy) [inline app/models/taxonomies/location.rb]: shouldnt it be written as:

``` ruby
conditions =  if user.admin?
               { }
   else
               sanitize_sql_for_conditions([" (taxonomies.id in (?))", user.location_and_child_ids])
   end
```

Comment (ohadlevy) [inline app/models/taxonomies/location.rb]: or as a oneliner

File: app/models/taxonomies/organization.rb [hunk 6]
```diff
@@ -14,14 +14,14 @@ class Organization < Taxonomy
...
-      else
-        conditions = sanitize_sql_for_conditions([" (taxonomies.id in (?))", user.organization_and_child_ids])
-      end
-      where(conditions)
-    }
+    user = User.current
+    if user.admin?
+      conditions = { }
+    else
+      conditions = sanitize_sql_for_conditions([" (taxonomies.id in (?))", user.organization_and_child_ids])
```
Comment (ohadlevy) [inline app/models/taxonomies/organization.rb]: ditto

File: db/seeds.d/07-config_templates.rb [hunk 7]
```diff
@@ -15,57 +15,57 @@
...
+    { :name => 'redhat_register', :source => 'snippets/_redhat_register.erb', :snippet => true },
+    { :name => 'saltstack_minion', :source => 'snippets/_saltstack_minion.erb', :snippet => true }
   ].each do |input|
     next if ConfigTemplate.find_by_name(input[:name])
     next if audit_modified? ConfigTemplate, input[:name]
     t = ConfigTemplate.create({
-                                  :snippet  => false,
-                                  :template => File.read(File.join("#{Rails.root}/app/views/unattended", input.delete(:source)))
-                              }.merge(input))
+      :snippet  => false,
```
Comment (ohadlevy) [inline db/seeds.d/07-config_templates.rb:66]: not sure i get this indentation rules, on a if condition we change indentation and on assignment we do not?

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (ohadlevy) [inline app/controllers/home_controller.rb:25]: i see, thanks.

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (ohadlevy) [inline app/helpers/application_helper.rb:213]: i see too, ignore :)

**Comment by ohadlevy** [comment]
also, i got the following errors when running rubocop manually:

```
config/environments/profile.rb:112:18: C: Indent the right bracket the same as the left bracket.
                 )
                 ^
config/environments/profile.rb:138:15: W: File.exists? is deprecated in favor of File.exist?.
      if File.exists?("#{engine.root}/public/assets")
              ^^^^^^^
app/models/hostgroup.rb:135:4: W: end at 135, 3 is not aligned with groups.each do |hg| at 133, 4
   end
   ^^^

```

File: app/models/compute_resources/foreman/model/rackspace.rb (hunk 4 above)
Comment (dLobatog) [inline app/models/compute_resources/foreman/model/rackspace.rb:108]: Most definitely yes, let me change it.

File: app/controllers/hosts_controller.rb (hunk 2 above)
Comment (dLobatog) [inline app/controllers/hosts_controller.rb:504]: I think in this case it makes the first few lines very long. In any case the cop this fixes had to do with the indentation of the `if else` block.

File: db/seeds.d/07-config_templates.rb (hunk 7 above)
Comment (dLobatog) [inline db/seeds.d/07-config_templates.rb:66]: This one in particular is fixing 

``` ruby
# C: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
                                   :snippet  => false,
#                                 ^^^^^^^^^^^^^^^^^^
# C: Indent the right brace the same as the start of the line where the left brace is.
                              }.merge(input))
```

File: app/models/taxonomies/location.rb (hunk 5 above)
Comment (dLobatog) [inline app/models/taxonomies/location.rb]: Okay, I'll change it to

``` ruby
      conditions = User.current.admin? ? {} : sanitize_sql_for_conditions([" (taxonomies.id in (?))", user.location_and_child_ids])
      where(conditions)
```

**Comment by dLobatog** [comment]
@ohadlevy config/environments/profile.rb is not a file of this repository, so it's :ok: if it fails rubocop :smile: , I've addressed the other comments.

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
@ohadlevy fixed the problem with hostgroup.rb, tests are :green_apple:

**Comment by ohadlevy** [comment]
[test]

**Comment by dLobatog** [comment]
@ohadlevy Rebased against latest develop, latest changes to config/apipie.rb broke it...

**Comment by dLobatog** [comment]
@ohadlevy Tests are :green_apple: now thankfully :smile:

**Comment by ohadlevy** [comment]
merged as 96144a4, thanks @eLobato !!

**Pull request closed without merging**